test/symgroups.js

```diff
@@ -0,0 +1,106 @@
+    var tex = fs.createWriteStream("symgroups.tex");
+    tex.on("finish", typeset);
+    tex.write("\\documentclass{article}\n" +
+              "\\usepackage{textcomp,amsmath,amssymb,gensymb}\n" +
+              "\\begin{document}\n" +
+              "\\showboxbreadth=\\maxdimen\\showboxdepth=\\maxdimen\n\n");
+    keys.forEach(function(key, idx) {
+        var sym = symbols.math[key];
+        var type = types.indexOf(sym.group) + 1;
+        tex.write("$" + idx + "+" + key + "+" + type + "\\showlists$\n\n");
+    });
+    tex.end("\\end{document}\n");
+}
+
+function typeset() {
+    var proc = childProcess.spawn(
+        "pdflatex", ["--interaction=nonstopmode", "symgroups"],
+        {stdio: "ignore"});
+    proc.on("exit", function(code, signal) {
+        if (signal) {
+            throw new Error("pdflatex terminated by signal " + signal);
+        }
+        fs.readFile("symgroups.log", "ascii", evaluate);
+    }).on("error", function(err) {
+        throw err;
+    });
+}
+
+var reMM = /^### math mode entered.*\n([^]*?)###/mg;
+var reParts = /([^]*^\.\\fam0 \+\n)([^]+)(\\mathbin\n\.+\\fam0 \+[^]*)/m;
+var reEmpty = /^\.\\fam0 \+\n\\mathbin\n\.\\fam0 \+/m;
+var reDigit = /^\.\\fam0 ([0-9])/mg;
+var reAtom = /\\([a-z]+)/;
+
+function evaluate(err, log) {
+    if (err) {
+        throw err;
+    }
+
+    var match, nextidx = 0;
+    while ((match = reMM.exec(log)) !== null) {
+        var list = match[1];
+        match = reParts.exec(list);
+        if (!match) {
+            match = reEmpty.exec(list);
+            if (match) {
+                console.log("Empty: ", nextidx, keys[nextidx]);
+                nextidx++;
+                continue;
+            }
+            console.error("Can't split this into parts:");
+            console.error(list);
+            process.exit(2);
+        }
+        var idx = extractDigits(match[1]);
+        var atom = match[2];
+        var expected = types[extractDigits(match[3]) - 1] || "???";
+        match = reAtom.exec(atom);
+        if (!match) {
+            console.error("Failed to find atom type");
+            console.error(atom);
+            console.error(list);
+            process.exit(3);
+        }
+        var actual = match[1];
+        if (expected !== actual && "math" + expected !== actual &&
+            (expected !== "textord" || actual !== "mathord")) {
+            console.log(expected, actual, idx, keys[idx]);
+            console.log(atom);
+        }
+        if (nextidx !== idx) {
+            console.error("Index " + nextidx + " not found in log");
+            process.exit(4);
+        }
+        nextidx = idx + 1;
+    }
+    if (nextidx !== keys.length) {
+        console.error("Processed " + nextidx +
+                      " out of " + keys.length + " symbols");
+        process.exit(4);
+    }
+}
+
+function extractDigits(str) {
+    var match, res = "";
+    while ((match = reDigit.exec(str)) !== null) {
+        res += match[1];
+    }
+    return +res;
```

Comment:
`parseInt('12hello', 10)` gives 12 so I tend to prefer unary `+` like this has. If you want to be more explicit, `Number(res)` is equivalent.